### ISSUE TITLE
SFT Scripts

### ISSUE THREAD

**Issue by ChanLiang** [zwhe99/DeepMath#2]
Thank you for your amazing work on DeepMath! I couldn't find the SFT scripts (e.g., using one and three R1 answers). Would you consider sharing this part?

Thank you!

**Comment by zwhe99** [comment]
We used the trainer from Transformers for SFT and here are all the parameters:

```shell
--batch_size 128 \
--micro_batch_size 1 \
--num_train_epochs 1 \
--learning_rate 1e-6 \
--lr_scheduler_type cosine \
--warmup_steps 50 \
--save_strategy epoch \
--bf16 \
--gradient_checkpointing \
--padding_free \
--cutoff_len 9216 \
--deepspeed deepspeed_config_zero3.json
```

**Comment by ChanLiang** [comment]
Thank you for your response and for providing the shell!

Regarding the SFT mentioned in the paper, does the "3 R1 Solutions" in Table 3 mean training for three epochs, with each epoch utilizing a different R1 response?

**Comment by zwhe99** [comment]
No. We just combined (x, y1), (x, y2), (x, y3), shuffled them and trained 1 epoch.

**Comment by zwhe99** [comment]
Closing this issue due to lack of activity. Don't hesitate to reopen it if you have any other questions.
